lnst/Recipes/ENRT/ConfigMixins/EnableIntelSSTTFMixin.py

```diff
@@ -0,0 +1,110 @@
+import re
+
+from lnst.Common.LnstError import LnstError
+from lnst.Common.Parameters import BoolParam
+from lnst.Controller.RecipeResults import ResultLevel, ResultType
+from lnst.Recipes.ENRT.ConfigMixins import BaseSubConfigMixin
+
+
+class EnableIntelSSTTFMixin(BaseSubConfigMixin):
+    enable_intel_sst_tf = BoolParam(default=False)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if getattr(self.params, "disable_turboboost", True):
+            raise ValueError(
+                "turboboost needs to be enabled when enabling intel_sst. "
+                f"intel_sst just allows higher clock speed for turbo boost."
+            )
+
+    @property
+    def intel_sst_tf_host_list(self):
+        """
+        The value of this property is a list of hosts for which the intel
+        SST-TF should be turned on. Derived class can override this property.
+        """
+        return self.matched
+
+    def _is_sst_supported(self, host):
+
+        sst_info = host.run("intel-speed-select --info", job_level=ResultLevel.DEBUG)
+
+        result = sst_info.passed
+        if not re.search(
+            r"Intel\(R\) SST-TF \(feature turbo-freq\) is supported", sst_info.stderr
+        ):
+            result = ResultType.FAIL
+
+        return result
+
+    def _system_cpus(self, host):
+        job = host.run(
+            "cat /proc/cpuinfo | grep -E '^processor' | cut -d':' -f2 | tr -d ' '"
+        )
+        if not job.passed:
+            raise LnstError("Could not get system cpus.")
+
+        return ",".join(cpu.strip() for cpu in job.stdout.splitlines())
+
+    def _ssted_cpus(self):
+        cpus = []
+        if "perf_tool_cpu" in self.params:
+            cpus.extend(self.params.perf_tool_cpu)
+
+        if "dev_intr_cpu" in self.params:
+            cpus.extend(self.params.dev_intr_cpu)
+
+        # TODO: multi dev intr config mixin support
+
+        return ",".join(str(cpu) for cpu in cpus)
+
+    def apply_sub_configuration(self, config):
+        super().apply_sub_configuration(config)
+
+        if not self.params.enable_intel_sst_tf:
+            return
+
+        for host in self.intel_sst_tf_host_list:
+            if not self._is_sst_supported(host):
+                raise LnstError(f"Machine {host} doesn't support Intel SST-TF.")
+
+            idlestates = host.run("cpupower idle-set -E")
+            if not idlestates.passed:
```

Comment:
I'll leave that up to you, but if you don't need the idlestates variable for other purpose, maybe just squash this into:
```python
if not host.run("cpupower idle-set -E").passed:
```

Applies to other run() calls below.